articles/building-a-python-package-and-publishing-on-pypi/index.md

```diff
@@ -0,0 +1,169 @@
+      'numpy',
+      'pandas'
+]
+
+setup(name=PACKAGE_NAME,
+      version=VERSION,
+      description=DESCRIPTION,
+      long_description=LONG_DESCRIPTION,
+      long_description_content_type=LONG_DESC_TYPE,
+      author=AUTHOR,
+      license=LICENSE,
+      author_email=AUTHOR_EMAIL,
+      url=URL,
+      install_requires=INSTALL_REQUIRES,
+      packages=find_packages()
+      )
+
+```
+
+The first chunk of code imports all the tools we need to set up our library, the *VERSION* will be the release number for the package, the *PACKAGE NAME* will be a unique name you want to call your package, the *AUTHOR* and *AUTHOR EMAIL* will be your details and finally, the URL can be a site if you have one or maybe your GitHub Repo.
+
+If your package requires other packages to work then you’ll need to indicate them, for example, if you use other packages to write your code you should add them in the *INSTALL_REQUIRES* because your package depends on them (otherwise it will not work) but this package we just created doesn’t require any and I intentionally put wrote this note, in case you’re trying it out with your package idea.
+
+#### Packaging
+Now that we are done writing the code, we will set up all the needed files, and structure the codebase. Another important step will be to sign up for an account on [pypi.org](pypi.org), because that is where we will be hosting our package. After you sign up go to your terminal and pip install Twine with the command below.
+
+```python
+pip install twine
+
+```
+
+The next thing you need to do is navigate to the directory of your project where the setup.py file is and then run this code.
+
+```python
+Python setup.py sdist bdist_wheel
+```
+
+After running the code above you will notice more directories called *dist*, *build*, and *your_package.egg-info*. The *dist* is the most important right now because it holds the installation files that we will deploy to PyPI, open the directory and you’ll see two compressed files that are saved with *.tar* extension and also a wheel file.
+
+#### Uploading package on PyPI
+Next, you will need to make sure that the distribution files that we created are working perfectly by running the code below.
+
+```python
+twine check dist/*
+```
+
+Now, let’s upload our package to PyPi but before we do that let’s deploy on a PyPI test domain so we can be sure that everything is running like it should be.
+
+To do that use the code below:
+
+```python
+twine upload --repository-url https://test.pypi.org/legacy/dist/*
+```
+
+To try out the test version you just created go to [test.pypi.org](https://test.pypi.org/) and check out the package, if you’re satisfied with what you see, then it is time to push it to the PyPi.
+
+```python
+Twine upload dist/*
+```
+
+Those are the steps required in building and deploying our package, you can run pip install <package name> to install it on your computer and try out the package using the code below.
+
+```python
+print(cal_lcm(4,5))
```

Comment:
Show the steps as follows;
1. Pip install calc_lcm
2. import calc_lcm
3. Then call the function.